koswat/strategies/infra_priority_strategy/infra_priority_strategy.py

```diff
@@ -0,0 +1,108 @@
+from collections import defaultdict
+from itertools import groupby
+
+from koswat.dike_reinforcements.reinforcement_profile.reinforcement_profile_protocol import (
+    ReinforcementProfileProtocol,
+)
+from koswat.strategies.order_strategy.order_strategy import OrderStrategy
+from koswat.strategies.strategy_input import StrategyInput
+from koswat.strategies.strategy_location_reinforcement import (
+    StrategyLocationReinforcement,
+)
+from koswat.strategies.strategy_protocol import StrategyProtocol
+
+
+class InfraPriorityStrategy(StrategyProtocol):
+    """
+    This strategy works under the assumption that
+    `StrategyLocationReinforcement.available_measures`
+    ARE NOT ordered by "compatibility".
+    Thus we need to check first which are the compatible reinforcements
+    per cluster.
```

Comment:
I think we may assume they are ordered by compatibility after #190. Should we add a subtask in that (or current) issue to remove superflous checks?
Btw, where can these compatibility be found?

Reply:
I think is best to create an issue for that, though I am not sure it is really needed here, as I prefer to work here under the assumption that we do not remove items from the `StrategyInput` properties.